### ISSUE TITLE
Upgrade to buildkit v0.11 

### ISSUE THREAD

**Issue by sipsma** [dagger/dagger#4293]
There's a lot of upstream fixes we've made to buildkit that we want to pick up (will link other issues to this).

It would be great to have this as part of the upcoming engine v0.3.8 release, so adding to that milestone, but it's not strictly necessary so can be pushed if v0.11 is not out by the time the other requirements for 0.3.8 are done.

**Comment by sipsma** [comment]
Did an initial upgrade to v0.11.0-rc3, so hopefully the final upgrade to the official v0.11 will be easy https://github.com/dagger/dagger/pull/4296

**Comment by sipsma** [comment]
Waiting for another rc or the official release

**Comment by sipsma** [comment]
We did our release w/ just rc3 since it didn't make sense to wait on buildkit any longer. Buildkit is planning to do a v0.11.1 release with some hot fixes. We should upgrade to that when it's out.

**Comment by TomChv** [comment]
[v0.11.2](https://github.com/moby/buildkit/releases) is live, shall we update it rn?

**Comment by sipsma** [comment]
We've already done this actually, I should have closed the issue a while ago, my bad 🙏
